lib/ansible/parsing/vault/__init__.py

```diff
@@ -528,6 +533,19 @@ def create_key(self, password, salt, keylength, ivlength):
         return derivedkey
 
     def gen_key_initctr(self, password, salt):
+        '''
+        Takes a password and random 32-byte salt and stretches it using
+        PBKDF2 into an 32-byte AES key, a 32-byte HMAC-SHA-256 key, and
+        a 16-byte IV for AES-CTR.
+
+        FIXME: There's really no reason to use PBKDF2 to generate the initial
+        counter value. It just slows things down (specifically, it adds 10000
+        iterations of the PRF, and then throws half the 32-byte output away).
+
+        CTR mode requires that the same key and the same counter are only ever
+        used once for encryption. Since we generate the key using a random salt
+        each time, it would be fine to always start the counter at 0.
```

Comment:
The "problem" with a random 16-byte (not bit) IV is that it'll have to be stored alongside the random 32-byte salt in the output. It wouldn't hurt anything, but since there's also no point to doing it, I don't prefer this solution.